app/src/main.rs

```diff
@@ -40,11 +47,15 @@ fn main(_argc: isize, _argv: *const *const u8) -> isize {
     batch_spawn_entities(&mut world, 5);
     let mut velocity_query = PreparedQuery::<&mut Velocity>::default();
     let mut all_query = PreparedQuery::<(&mut Position, &mut Velocity)>::default();
-    
+
     // Kickstart main loop.
     let mut renderer = Renderer::new();
     renderer.init_render();
+    let mut last_frame_time = ogc_rs::system::System::system_time();
     loop {
+        let now = ogc_rs::system::System::system_time();
+        let delta_time = now - last_frame_time;
```

Comment:
This might be a bit silly, but what about creating a `Time`  component, with a ` system_update_time`  system function that queries all entities with a `Time` component and sets it with the delta time of this main loop?

That way other queries can use the delta time by simply requesting e.g. ` Time`  and `Velocity`  at the same time, then doing the math. Or... have a ` system_update_modulators`  system function that queries both `Time` and `Modulator` components :stuck_out_tongue: .

It might seem early to talk about this, but I' d like to keep the example app in this repo in a shape that will allow us to easily start building on it. So, I'm somewhat aiming for a vague balance between "meh, it' s a demo" and "here's your framework" haha

Reply:
What seemed like a better abstraction to me, is to essentially have the following:

```rust
pub struct GameState {
  world: hecs::World,
  // other global game data like score, toggle between modes, etc.
}

pub struct Changes {
  pub controls: Controls,
  pub delta_time_ms: u64,
  // Other things we expect the game loop/Wii hardware to provide each frame
}

pub struct Controls {
  home_button_down: bool,
  one_button_down: bool,
  // Etc. Any current controller state we are interested in including where the wiimote(s) point.
}

impl GameState {
  /// Set up anything required for a new game
  pub fn new() -> GameState {
    // ...
  }

   /// Performs game logic of one iteration of the game loop.
  /// Before calling this, the caller should have filled `Changes` with the required values.
  /// 
  /// If the returned bool is `true`, the game loop should continue. 
  /// When `false` is returned, the game should quit.
  pub fn update(&mut GameState, &Changes) -> bool {
    // ...
  }
}
```

Reply:
I like this better as well, good thinking as usual. Could you make a ticket for this implementation and point to this comment?

Reply:
I made #31 for this